src/types/integer.rs

```diff
@@ -0,0 +1,201 @@
+use crate::result::{decoding_error, IonError};
```

Comment:
PR tour: everything in this file was moved wholesale from the `value` module when I promoted `AnyInt` to a public-facing type. There are no logic changes.